Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -2317,22 +2317,16 @@ static bool shouldPrependSpace(AXCoreObject& object, AXCoreObject* previousObjec
 
 String AccessibilityNodeObject::textUnderElement(TextUnderElementMode mode) const
 {
-    auto isAriaVisible = [this] () {
-        return Accessibility::findAncestor<AccessibilityObject>(*this, true, [] (const auto& object) {
-            return equalLettersIgnoringASCIICase(object.getAttribute(aria_hiddenAttr), "false"_s);
-        }) != nullptr;
-    };
-
     RefPtr node = this->node();
     if (auto* text = dynamicDowncast<Text>(node.get()))
-        return !mode.isHidden() || isAriaVisible() ? text->wholeText() : emptyString();
+        return !mode.isHidden() && !isAXHidden() ? text->wholeText() : emptyString();
 
     const auto* style = this->style();
     mode.inHiddenSubtree = WebCore::isDOMHidden(style);
     // The Accname specification states that if the current node is hidden, and not directly
     // referenced by aria-labelledby or aria-describedby, and is not a host language text
     // alternative, the empty string should be returned.
-    if (mode.isHidden() && !isAriaVisible() && (node && !ancestorsOfType<HTMLCanvasElement>(*node).first())) {
+    if ((mode.isHidden() || isAXHidden()) && (node && !ancestorsOfType<HTMLCanvasElement>(*node).first())) {
```

Comment:
I built this patch and removed the`|| isAXHidden()` check you added here, and no tests failed. Do we actually need this?